spec/controllers/submissions_controller_spec.rb

```diff
@@ -121,6 +121,68 @@
       end
     end
 
+    context 'submitting a url' do
+      describe 'should add url files' do
+        it 'returns ok response' do
+          post_as @student, :update_files,
+                  params: { course_id: course.id, assignment_id: @assignment.id,
+                            new_url: 'https://www.youtube.com/watch?v=dtGs7Fy8ISo', url_text: 'youtube' }
+          expect(response).to have_http_status :ok
+        end
+
+        it 'added a new file' do
+          post_as @student, :update_files,
+                  params: { course_id: course.id, assignment_id: @assignment.id,
+                            new_url: 'https://www.youtube.com/watch?v=dtGs7Fy8ISo', url_text: 'youtube' }
+          @grouping.group.access_repo do |repo|
+            revision = repo.get_latest_revision
+            files = revision.files_at_path(@assignment.repository_folder)
+            expect(files['youtube.mkurl']).to_not be_nil
+          end
+        end
+      end
+
+      describe 'should reject url with no name' do
+        it 'returns a bad request' do
+          post_as @student, :update_files,
+                  params: { course_id: course.id, assignment_id: @assignment.id,
+                            new_url: 'https://www.youtube.com/watch?v=dtGs7Fy8ISo' }
+          expect(response).to have_http_status :bad_request
+        end
+
+        it 'does not add a new file' do
+          post_as @student, :update_files,
+                  params: { course_id: course.id, assignment_id: @assignment.id,
+                            new_url: 'https://www.youtube.com/watch?v=dtGs7Fy8ISo' }
+          @grouping.group.access_repo do |repo|
+            revision = repo.get_latest_revision
+            files = revision.files_at_path(@assignment.repository_folder)
+            expect(files['youtube.mkurl']).to be_nil
+          end
+        end
+      end
+
+      describe 'should reject url with no name' do
```

Comment:
This tests: `'should reject url with no name'` but both tests send requests with `url_text` as a parameter

Reply:
This test should read `should reject invalid url`. Thanks for the catch.